### PR TITLE
CTCP-1970: AddAnotherDocument page

### PR DESCRIPTION
**Pull request by kevincoatsworth**

### CODE REVIEW

File: app/controllers/AddAnotherDocumentController.scala [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  def onSubmit(lrn: LocalReferenceNumber, mode: Mode): Action[AnyContent] = actions.requireData(lrn) {
+    implicit request =>
+      val viewModel = viewModelProvider(request.userAnswers, mode)
+      form(viewModel)
+        .bindFromRequest()
+        .fold(
+          formWithErrors => BadRequest(view(formWithErrors, lrn, viewModel)),
+          {
+            case true  => Redirect(controllers.document.routes.TypeController.onPageLoad(lrn, NormalMode, viewModel.nextIndex))
```
Comment (jackheslop96) [inline app/controllers/AddAnotherDocumentController.scala:70]: Is this correct?

File: conf/messages.en [hunk 2]
```diff
@@ -17,6 +17,14 @@ error.summary.title = There is a problem
 
 document.caption = Documents
 
+addAnotherDocument.singular.title = You have added {0} document
+addAnotherDocument.singular.heading = You have added {0} document
+addAnotherDocument.plural.title = You have added {0} documents
+addAnotherDocument.plural.heading = You have added {0} documents
+addAnotherDocument.error.required = Select yes if You have added 1 document
+addAnotherDocument.label = Do you want to add another document?
+addAnotherDocument.maxLimit.label = You cannot add any more documents. To add another document, you need to remove one first.
```
Comment (jackheslop96) [inline conf/messages.en:26]: These may have already been defined. Feel free to remove the other ones.

File: app/controllers/AddAnotherDocumentController.scala (hunk 1 above)
Comment (kevincoatsworth) [reply app/controllers/AddAnotherDocumentController.scala:70]: I believe so, the onPageLoad would redirect to the relevant starting page if they have 0 documents added. If they land on the page, they must at least have 1 document added and after the first index it's always the Type page

Comment (jackheslop96) [reply app/controllers/AddAnotherDocumentController.scala:70]: Ah yes, I see that on the drawio now. Good shout. Marking this as resolved.

**Review by jackheslop96 — APPROVED** [review]

**Pull request merged**